### PR TITLE
Allow for mixing parameter styles

### PR DESCRIPTION
**Pull request by ovadbar**
This should allow for mixed styles of parameters.

### CODE REVIEW

File: internal/endtoend/testdata/mix/test.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	id serial not null,
+	name text not null,
+	phone text not null
+);
+
+-- name: CountOne :one
+SELECT count(1) FROM bar WHERE id = $2 AND phone < @phone_param and name <> $1;
+
+-- name: CountTwo :one
+SELECT count(1) FROM bar WHERE id = sqlc.arg(id_param) AND phone < @phone_param and name <> $1;
```
Comment (kyleconroy) [inline internal/endtoend/testdata/mix/test.sql:11]: The @param style will be deprecated soon (see #605) so let's remove them from the test.

File: internal/endtoend/testdata/mix/sqlc.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "version": "1",
+  "packages": [
+    {
+      "path": "go",
+      "name": "querytest",
+      "schema": "test.sql",
```
Comment (kyleconroy) [inline internal/endtoend/testdata/mix/sqlc.json:7]: Please add the "engine" key and set it `postgresql`.

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Thanks for the patch! I agree that it doesn't make much sense to require a single parameter style. A few request:

* please pick a more descriptive name for the tests, such as `sqlc_arg_mixed` or `params_mixed`.
* Move the current tests into a `postgresql` subdirectory and add the same test for MySQL

**Comment by ovadbar** [comment]
Thanks for the feedback.

(One thing I did notice is if you have a query as follows, the numbers always precede the sqlc.args convetion)

```
SELECT count(1) FROM table_name WHERE id > $1 AND id < sqlc.arg(id);
```

**Comment by dant4301** [comment]
This would be great fix and would be awesome if merged.

There is an outstanding issue w.r.t using named_parameter for LIMIT for MySQL. We can use this fix to sort of work around it in the short term, but for the below:

I applied the patch and tried it for the query:
```
SELECT * FROM books WHERE book_id >= sqlc.arg(min_book_id) ORDER BY title ASC LIMIT ?;
```
The generated code looks like this:
```
type GetBooksByNameParams struct {
        Limit                 int32 `json:"limit"`
        MinBookID int64 `json:"min_book_id"`
}

func (q *Queries) GetBooksByName(ctx context.Context, arg GetBooksByNameParams) ([]Category, error) {
        rows, err := q.query(ctx, q.getBooksByNameStmt, getCategoriesByName, arg.Limit, arg.MinBookID)
...
```
Aren't the orders of parameters arg.Limit and arg.MinBookID switched?

**Comment by ovadbar** [comment]
Yeah I kind of realized that was an issue but, with postgresql you can use dollar numbers format so it is not an issue and I did not think of how it would work with mysql.

I can look into fixing it but I really do not have the time.

**Comment by kyleconroy** [comment]
> I can look into fixing it but I really do not have the time.

Understandable. I also don't have the time to look into fixing this. @dant4301 If you'd like this feature, you're going to have to get your hands dirty and take a shot at fixing this PR. I'm going to close this for now, but feel free to fork this branch and open up another PR if you can get it to work.

**Comment by ovadbar** [comment]
@dant4301 can you look at https://github.com/kyleconroy/sqlc/pull/1071 and see if the new pull request solves your problems

**Comment by dant4301** [comment]
@ovadbar,  I do not use sqlc now since I did a limited amount of codegen just enough for my needs. Thanks again though.

**Pull request closed without merging**